hostfactory/host_provider/src/cluster.py

```diff
@@ -18,31 +19,68 @@ class AutoscalingStrategy(enumerate):
     CAPACITY = "capacity"
     WEIGHTED = "weighted"
 
-# This is only for unit testing
-def allocation_strategy(node_mgr, template_id, slot_count, capacity_limit_timeout, vm_sizes):
-    per_vm_size = slot_count//len(vm_sizes)
-    remainder = slot_count % len(vm_sizes)
-    print(vm_sizes)
+def allocation_strategy(node_mgr, logger, template_id, slot_count, capacity_limit_timeout, vm_sizes, vm_dist):
+    result = None
+    check_allocate = None
+    logger.debug(f"vm dist {vm_dist}")
```

Comment:
Try to expand on this log statement - "Allocating {slot_count} slots for template_id {template_id} using weights {vm_dist}"
or something along those lines. You may want to format the vm_dist rather than relying on the default __str__ implementation